packages/plugins/prometheus/src/utils.ts

```diff
@@ -89,14 +89,14 @@ export type ShouldObservePredicate<Params extends Record<string, any>> = (
 ) => boolean;
 
 export type HistogramAndLabels<
-  Phases extends string[],
+  Phases,
   LabelNames extends string,
   Params extends Record<string, any>,
 > = {
   histogram: Histogram<LabelNames>;
   fillLabelsFn: FillLabelsFn<LabelNames, Params>;
-  phases: Phases;
-  shouldObserve: ShouldObservePredicate<Params>;
+  phases?: AtLeastOne<Phases>;
+  shouldObserve?: ShouldObservePredicate<Params>;
```

Comment:
Breaking change?

Reply:
Nope, it's a new thing from the previous PR.
Perhaps it would be clearer to review if I revert the merge commit from previous PR and cherry-pick it in this one ?